### PR TITLE
Dequeue all messages

### PR DESCRIPTION
**Pull request by pmengelbert**
*relevant diff here:* https://github.com/Azure/moby-packaging/compare/skip_when_no_messages...dequeue_all_messages

*note* this PR is based on #96, and that PR should be merged first.

- Move message fetching to go program
- Fetch all messages before proceeding, instead of a batch of 32

### CODE REVIEW

**Comment by cpuguy83** [comment]
Needs a rebase.

**Comment by pmengelbert** [comment]
> Needs a rebase.

@cpuguy83 Done, thanks

File: queue_consumer.yaml [hunk 1]
```diff
@@ -73,7 +67,7 @@ stages:
                 'https://dev.azure.com/AzureContainerUpstream/Moby/_apis/build/builds/$(Build.BuildId)?api-version=7.0'
             env:
               MSG_DIR: "$(Pipeline.Workspace)/messages"
-              UPDATED_BUILD_NUMBER: "${{ parameters.build_id }}.no.new.queue.messages"
+              UPDATED_BUILD_NUMBER: "${{ replace(parameters.build_id, '__not_supplied', '$(Date:yyyyMMdd)') }}.no.new.queue.messages"
```
Comment (cpuguy83) [inline queue_consumer.yaml:70]: Looks like the pipeline errored here.

Comment (pmengelbert) [reply queue_consumer.yaml]: Reverted for now. I'll revisit later

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**